### ISSUE TITLE
Postman Mac App Hangs for several minutes when trying to move requests between collecitons

### ISSUE THREAD

**Issue by btwynham** [a85/POSTMan-Chrome-Extension#3776]
##
Postman Helper goes to 100% CPU usage and Postman app/Gui hangs for several minutes when I move a request between a collection

## App Details:
Postman for Mac
Version 5.3.2
OS X 15.6.0 / x64

## Issue Report:
 1. I started seeing this issue more and more over the past month I did not initially have this issue.
 2. I should be able to move requests across collections with out seeing this issue
Once my App "recovers" and stops hanging I am able to move requests across collections it is like the first one is causing some issue and after that the issue goes away.
 3. Console logs 
Check Console logs but they don't seem to show anything
 4. Screenshots (if applicable)

Happy to collect proper logs to help debug this issue.

This issue happens every time for me.

I have about 30 collections with 100 or so requests.
Don't know if the Postman App is trying to access them all at the same time when I try and move a request.

<img width="1699" alt="screen shot 2017-11-10 at 1 49 38 pm" src="https://user-images.githubusercontent.com/33555861/32673875-b9b0386c-c61e-11e7-9e0d-3218aec19a03.png">

**Comment by sdnts** [comment]
Hi @btwynham Is the app in a `Syncing` state when it hangs? 3000 requests is not a lot of requests, so this sounds like a performance problem.

**Comment by btwynham** [comment]
No I don't believe the App is in syncing state, but I am happy to check.

I installed the postman canary which i think is more streamlined and I can move requests just fine with that.

I have a pretty beffy mac, 16 GM of Ram and 2.5Ghz Intel Core i7 processor so I don't think it is a performance issue on my end. This is usually the only app running in addition to google chrome.

Could their be a contention between Google Chrome and Postman?

**Comment by btwynham** [comment]
I took two samples of my Mac when it gets stuck and when it isn't stuck.

There are lot more nested electron framework when it gets stuck maybe a nested loop there?

[SampleWhenNotStuck.txt](https://github.com/postmanlabs/postman-app-support/files/1483872/SampleWhenNotStuck.txt)
[SampleWhenStuck.txt](https://github.com/postmanlabs/postman-app-support/files/1483873/SampleWhenStuck.txt)

**Comment by btwynham** [comment]
I must have hadd about 100 Collections.

I saved my collections off and then deleted them.

And reduced my collections to a total of 12.

Once i did this I no longer have the issue of moving requests.

So I think you guys probably have a scalability issue.

I had thought it had to do with your sync feature but I turned that off and I still ahd the issue.

So maybe when I go to move a request I think you try to load all of the json structures into memory.
